app/models/logout_utility.rb

```diff
@@ -60,5 +60,27 @@ def build_request(id_token:, post_logout_redirect_uri:, state: nil)
           )
         end
       end
+
+      def check_initialize_arguments!(idp_base_url, idp_configuration, end_session_endpoint)
```

Comment:
Looks like this method and build request is repeated. Also, please make sure there are correct ends